docs/hashing.rst

```diff
@@ -50,6 +53,16 @@ Because according to the definition_ from the official Python docs, the returned
 For a more thorough explanation of this topic, please refer to this blog post: `Python Hashes and Equality`_.
 
 
+Hash Code Caching
+-----------------
+
+Some objects have hash codes which are expensive to compute.
+If such objects are to be stored in hash-based collections, it can be useful to compute the hash codes only once and then store the result on the object to make future hash code requests fast.
+To enable caching of hash codes, specify ``cache_hash=True``.
+This may only be done if ``attrs`` is already generating a hash function for the object.
+If the hash code is cached, no field involved in hash code computation may be mutated after construction.
+It is strongly recommended that classes with cached hashcodes be ``frozen``
```

Comment:
Period at the end. Maybe start sentence with `Therefore`?

Also thinking about this…it’s kind true for all hashes except that with caching, they mustn’t change after first hashing while with normal hashes they mustn’t change after been inserted somewhere. Is there a way to communicate that?

Reply:
@hynek : I tried adding the following section:

```
Hashing and Mutability
----------------------
Changing any field involved in hash code computation after the first call to `__hash__` (typically this would be after its insertion into a hash-based collection) can result in silent bugs.
Therefore, it is strongly recommended that hashable classes be ``frozen``.
```

This doesn't point out that it would be okay to modify the fields after calling `__hash__` but before collection insertion is the hash code is not cached, but this seems a weird enough (and dangerous enough) case that it should probably be avoided by users anyway.